src/Utilities/objective_container.jl

```diff
@@ -116,26 +116,58 @@ function MOI.supports(
     return true
 end
 
+function _get_single_variable(o::ObjectiveContainer{T}) where {T}
+    return o.single_variable::MOI.VariableIndex
+end
+
+function _get_scalar_affine(o::ObjectiveContainer{T}) where {T}
+    return o.scalar_affine::MOI.ScalarAffineFunction{T}
+end
+
+function _get_scalar_quadratic(o::ObjectiveContainer{T}) where {T}
+    return o.scalar_quadratic::MOI.ScalarQuadraticFunction{T}
+end
+
+function _get_scalar_nonlinear(o::ObjectiveContainer{T}) where {T}
+    return o.scalar_nonlinear::MOI.ScalarNonlinearFunction
+end
+
+function _get_vector_variables(o::ObjectiveContainer{T}) where {T}
+    return o.vector_variables::MOI.VectorOfVariables
+end
+
+function _get_vector_affine(o::ObjectiveContainer{T}) where {T}
+    return o.vector_affine::MOI.VectorAffineFunction{T}
+end
+
+function _get_vector_quadratic(o::ObjectiveContainer{T}) where {T}
+    return o.vector_quadratic::MOI.VectorQuadraticFunction{T}
+end
+
+function _get_vector_nonlinear(o::ObjectiveContainer{T}) where {T}
+    return o.vector_nonlinear::MOI.VectorNonlinearFunction
+end
+
 function MOI.get(
     o::ObjectiveContainer{T},
     ::MOI.ObjectiveFunction{F},
 ) where {T,F}
     if o.scalar_affine !== nothing
-        return convert(F, o.scalar_affine)
+        return convert(F, _get_scalar_affine(o))
```

Comment:
All the changes in this file deal with the problem that `o.scalar_affine::Union{Nothing,MOI.ScalarAffineFunction{T}}`.

At runtime, the `if` statement handles the `nothing` case, so we never call `convert(F, nothing)`, but it's useful to have a type-stable way to get the non-nothing function.

Although writing this now, I wonder if I could use `_not_nothing(o.scalar_affine)` to disambiguate the union.